src/NuGet.Core/NuGet.LibraryModel/CentralPackageVersion.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Text;
+using NuGet.Shared;
+using NuGet.Versioning;
+
+namespace NuGet.LibraryModel
+{
+    public class CentralPackageVersion : IEquatable<CentralPackageVersion>
+    {
+        public string Name { get; }
+
+        public VersionRange VersionRange { get; }
+
+        public CentralPackageVersion(
+            string name,
+            VersionRange versionRange)
+        {
+            Name = name;
+            VersionRange = versionRange;
+        }
+
+        public override string ToString()
+        {
+            var sb = new StringBuilder();
+            sb.Append(Name);
+            sb.Append(" ");
+            sb.Append(VersionRange);
+            return sb.ToString();
+        }
+
+        public override int GetHashCode()
+        {
+            var hashCode = new HashCodeCombiner();
+
+            hashCode.AddObject(Name.ToLowerInvariant());
+            hashCode.AddObject(VersionRange);
+
+            return hashCode.CombinedHash;
+        }
```

Comment:
Given this class is immutable, you could consider having the string and hash code in `Lazy<>` fields, so that if the method is called more than once, it doesn't need to do the CPU work to re-calculate the same value. Trade off memory for CPU. Probably less important for ToString, but GetHashCode has more potential for being called multiple times.